[
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1735932639",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1735932639",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1735932639,
        "node_id": "IC_kwDOBWUGps5neDrf",
        "user": {
            "login": "davidwendt",
            "id": 45795991,
            "node_id": "MDQ6VXNlcjQ1Nzk1OTkx",
            "avatar_url": "https://avatars.githubusercontent.com/u/45795991?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/davidwendt",
            "html_url": "https://github.com/davidwendt",
            "followers_url": "https://api.github.com/users/davidwendt/followers",
            "following_url": "https://api.github.com/users/davidwendt/following{/other_user}",
            "gists_url": "https://api.github.com/users/davidwendt/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/davidwendt/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/davidwendt/subscriptions",
            "organizations_url": "https://api.github.com/users/davidwendt/orgs",
            "repos_url": "https://api.github.com/users/davidwendt/repos",
            "events_url": "https://api.github.com/users/davidwendt/events{/privacy}",
            "received_events_url": "https://api.github.com/users/davidwendt/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2023-09-26T16:54:42Z",
        "updated_at": "2023-09-26T16:54:42Z",
        "author_association": "CONTRIBUTOR",
        "body": "I'm not able to reproduce the error on my local libcudf build.\r\n```\r\n$ /usr/local/cuda/bin/compute-sanitizer --version\r\nNVIDIA (R) Compute Sanitizer\r\nCopyright (c) 2020-2023 NVIDIA Corporation\r\nVersion 2022.4.1\r\n```\r\n```\r\n$ nvcc --version\r\nnvcc: NVIDIA (R) Cuda compiler driver\r\nCopyright (c) 2005-2022 NVIDIA Corporation\r\nBuilt on Mon_Oct_24_19:12:58_PDT_2022\r\nCuda compilation tools, release 12.0, V12.0.76\r\nBuild cuda_12.0.r12.0/compiler.31968024_0\r\n```",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1735932639/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1735941711",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1735941711",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1735941711,
        "node_id": "IC_kwDOBWUGps5neF5P",
        "user": {
            "login": "davidwendt",
            "id": 45795991,
            "node_id": "MDQ6VXNlcjQ1Nzk1OTkx",
            "avatar_url": "https://avatars.githubusercontent.com/u/45795991?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/davidwendt",
            "html_url": "https://github.com/davidwendt",
            "followers_url": "https://api.github.com/users/davidwendt/followers",
            "following_url": "https://api.github.com/users/davidwendt/following{/other_user}",
            "gists_url": "https://api.github.com/users/davidwendt/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/davidwendt/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/davidwendt/subscriptions",
            "organizations_url": "https://api.github.com/users/davidwendt/orgs",
            "repos_url": "https://api.github.com/users/davidwendt/repos",
            "events_url": "https://api.github.com/users/davidwendt/events{/privacy}",
            "received_events_url": "https://api.github.com/users/davidwendt/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2023-09-26T17:00:44Z",
        "updated_at": "2023-09-26T17:36:12Z",
        "author_association": "CONTRIBUTOR",
        "body": "We have a few reductions tests in libcudf that use min-aggregation followed by a call to max-aggregation. Here is one that executes with `int16` types:\r\nhttps://github.com/rapidsai/cudf/blob/3196f6c36140962818aa8d12fe4fbd0dc522e31e/cpp/tests/reductions/reduction_tests.cpp#L152-L157\r\nAll of the tests are run with compute-sanitizer in our nightly builds\r\nCurious if these tests also fail for you in your environment as well.",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1735941711/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1737442729",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1737442729",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1737442729,
        "node_id": "IC_kwDOBWUGps5nj0Wp",
        "user": {
            "login": "ttnghia",
            "id": 7416935,
            "node_id": "MDQ6VXNlcjc0MTY5MzU=",
            "avatar_url": "https://avatars.githubusercontent.com/u/7416935?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/ttnghia",
            "html_url": "https://github.com/ttnghia",
            "followers_url": "https://api.github.com/users/ttnghia/followers",
            "following_url": "https://api.github.com/users/ttnghia/following{/other_user}",
            "gists_url": "https://api.github.com/users/ttnghia/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/ttnghia/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/ttnghia/subscriptions",
            "organizations_url": "https://api.github.com/users/ttnghia/orgs",
            "repos_url": "https://api.github.com/users/ttnghia/repos",
            "events_url": "https://api.github.com/users/ttnghia/events{/privacy}",
            "received_events_url": "https://api.github.com/users/ttnghia/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2023-09-27T13:50:21Z",
        "updated_at": "2023-09-27T13:50:21Z",
        "author_association": "CONTRIBUTOR",
        "body": "I suspect that this is the same as my previous reported issue: https://github.com/rapidsai/cudf/issues/13685",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1737442729/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1737453350",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1737453350",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1737453350,
        "node_id": "IC_kwDOBWUGps5nj28m",
        "user": {
            "login": "res-life",
            "id": 8166419,
            "node_id": "MDQ6VXNlcjgxNjY0MTk=",
            "avatar_url": "https://avatars.githubusercontent.com/u/8166419?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/res-life",
            "html_url": "https://github.com/res-life",
            "followers_url": "https://api.github.com/users/res-life/followers",
            "following_url": "https://api.github.com/users/res-life/following{/other_user}",
            "gists_url": "https://api.github.com/users/res-life/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/res-life/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/res-life/subscriptions",
            "organizations_url": "https://api.github.com/users/res-life/orgs",
            "repos_url": "https://api.github.com/users/res-life/repos",
            "events_url": "https://api.github.com/users/res-life/events{/privacy}",
            "received_events_url": "https://api.github.com/users/res-life/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2023-09-27T13:55:56Z",
        "updated_at": "2023-09-27T13:55:56Z",
        "author_association": "CONTRIBUTOR",
        "body": "> Curious if these tests also fail for you in your environment as well.\r\n\r\nYes, they also fail.\r\nI copied the `reduction_tests.cpp` file to [spark-rapids-jni](https://github.com/NVIDIA/spark-rapids-jni/tree/branch-23.10/src/main/cpp/tests), and compiled to an executable.\r\nTried to directly compile against the cuDF code, but the compile failed. So I copied the `reduction_tests.cpp` to spark-rapids-jni project and compile.",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1737453350/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1740109101",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1740109101",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1740109101,
        "node_id": "IC_kwDOBWUGps5nt_Ut",
        "user": {
            "login": "davidwendt",
            "id": 45795991,
            "node_id": "MDQ6VXNlcjQ1Nzk1OTkx",
            "avatar_url": "https://avatars.githubusercontent.com/u/45795991?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/davidwendt",
            "html_url": "https://github.com/davidwendt",
            "followers_url": "https://api.github.com/users/davidwendt/followers",
            "following_url": "https://api.github.com/users/davidwendt/following{/other_user}",
            "gists_url": "https://api.github.com/users/davidwendt/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/davidwendt/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/davidwendt/subscriptions",
            "organizations_url": "https://api.github.com/users/davidwendt/orgs",
            "repos_url": "https://api.github.com/users/davidwendt/repos",
            "events_url": "https://api.github.com/users/davidwendt/events{/privacy}",
            "received_events_url": "https://api.github.com/users/davidwendt/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2023-09-28T23:08:58Z",
        "updated_at": "2023-09-28T23:08:58Z",
        "author_association": "CONTRIBUTOR",
        "body": "This seems specific to your test environment since our nightly compute-sanitizer does not fail running `reduction_tests.cpp` and I cannot reproduce the error locally either.\r\n\r\nPerhaps you can provide some details on the environment. \r\nIs libcudf.so being built from source or downloaded from conda, etc?\r\nIs there a way to duplicate an environment where we could debug libcudf (e.g. compile and link specific modules)?\r\n\r\nI see mention of a docker image in the description. Does the error occur only on centos7?\r\nThis kind of error would not be silent and so should be occurring always. Does it show up in your normal testing?\r\nIn other words, does the error occur without running compute-sanitizer? \r\nIf not, perhaps this could be a compute-sanitizer bug on CUDA-12 on centos7 instead.",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1740109101/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1743119623",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1743119623",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1743119623,
        "node_id": "IC_kwDOBWUGps5n5eUH",
        "user": {
            "login": "jlowe",
            "id": 1360766,
            "node_id": "MDQ6VXNlcjEzNjA3NjY=",
            "avatar_url": "https://avatars.githubusercontent.com/u/1360766?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/jlowe",
            "html_url": "https://github.com/jlowe",
            "followers_url": "https://api.github.com/users/jlowe/followers",
            "following_url": "https://api.github.com/users/jlowe/following{/other_user}",
            "gists_url": "https://api.github.com/users/jlowe/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/jlowe/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/jlowe/subscriptions",
            "organizations_url": "https://api.github.com/users/jlowe/orgs",
            "repos_url": "https://api.github.com/users/jlowe/repos",
            "events_url": "https://api.github.com/users/jlowe/events{/privacy}",
            "received_events_url": "https://api.github.com/users/jlowe/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2023-10-02T14:24:57Z",
        "updated_at": "2023-10-02T14:24:57Z",
        "author_association": "MEMBER",
        "body": "> Is libcudf.so being built from source or downloaded from conda, etc? \r\n\r\nThis is a custom build of libcudf for the RAPIDS Accelerator, where we are compiling libcudf as a PIC static library that is ultimately linked into a shared library and used by the JVM.\r\n\r\n> Is there a way to duplicate an environment where we could debug libcudf (e.g. compile and link specific modules)?\r\n> I see mention of a docker image in the description. Does the error occur only on centos7?\r\n\r\nThe Docker image is used to produce this build, see https://github.com/NVIDIA/spark-rapids-jni/blob/branch-23.10/CONTRIBUTING.md#building-in-the-docker-container.  After pulling the spark-rapids-jni and executing the build-in-docker script, the libcudf install will be in spark-rapids-jni/target/libcudf-install/.  You can use the run-in-docker script to get an interactive shell within the same environment as the build env if desired.\r\n\r\nThe result of the build can be run on any supported OS (e.g.: Ubuntu).  I don't know if the error has been reproduced in different OS's.\r\n\r\n> This kind of error would not be silent and so should be occurring always. Does it show up in your normal testing?\r\n\r\nNo, it only occurs when running under the compute-sanitizer, and specifically when compiling with CUDA12.  I agree that at this point it appears to be a compute-sanitizer bug specific to CUDA 12.\r\n\r\n",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1743119623/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1804284370",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1804284370",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1804284370,
        "node_id": "IC_kwDOBWUGps5rizHS",
        "user": {
            "login": "GregoryKimball",
            "id": 12725111,
            "node_id": "MDQ6VXNlcjEyNzI1MTEx",
            "avatar_url": "https://avatars.githubusercontent.com/u/12725111?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/GregoryKimball",
            "html_url": "https://github.com/GregoryKimball",
            "followers_url": "https://api.github.com/users/GregoryKimball/followers",
            "following_url": "https://api.github.com/users/GregoryKimball/following{/other_user}",
            "gists_url": "https://api.github.com/users/GregoryKimball/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/GregoryKimball/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/GregoryKimball/subscriptions",
            "organizations_url": "https://api.github.com/users/GregoryKimball/orgs",
            "repos_url": "https://api.github.com/users/GregoryKimball/repos",
            "events_url": "https://api.github.com/users/GregoryKimball/events{/privacy}",
            "received_events_url": "https://api.github.com/users/GregoryKimball/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2023-11-09T17:46:55Z",
        "updated_at": "2023-11-09T17:46:55Z",
        "author_association": "CONTRIBUTOR",
        "body": "> I agree that at this point it appears to be a compute-sanitizer bug specific to CUDA 12.\r\nIf true, then the action here will be to file an nvbug with the compute-sanitizer team.",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1804284370/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    },
    {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1941133642",
        "html_url": "https://github.com/rapidsai/cudf/issues/14192#issuecomment-1941133642",
        "issue_url": "https://api.github.com/repos/rapidsai/cudf/issues/14192",
        "id": 1941133642,
        "node_id": "IC_kwDOBWUGps5zs1lK",
        "user": {
            "login": "davidwendt",
            "id": 45795991,
            "node_id": "MDQ6VXNlcjQ1Nzk1OTkx",
            "avatar_url": "https://avatars.githubusercontent.com/u/45795991?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/davidwendt",
            "html_url": "https://github.com/davidwendt",
            "followers_url": "https://api.github.com/users/davidwendt/followers",
            "following_url": "https://api.github.com/users/davidwendt/following{/other_user}",
            "gists_url": "https://api.github.com/users/davidwendt/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/davidwendt/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/davidwendt/subscriptions",
            "organizations_url": "https://api.github.com/users/davidwendt/orgs",
            "repos_url": "https://api.github.com/users/davidwendt/repos",
            "events_url": "https://api.github.com/users/davidwendt/events{/privacy}",
            "received_events_url": "https://api.github.com/users/davidwendt/received_events",
            "type": "User",
            "site_admin": false
        },
        "created_at": "2024-02-13T10:38:05Z",
        "updated_at": "2024-02-13T10:38:05Z",
        "author_association": "CONTRIBUTOR",
        "body": "Curious if this is perhaps resolved with the fixes for this issue https://github.com/NVIDIA/spark-rapids-jni/issues/1567 ",
        "reactions": {
            "url": "https://api.github.com/repos/rapidsai/cudf/issues/comments/1941133642/reactions",
            "total_count": 0,
            "+1": 0,
            "-1": 0,
            "laugh": 0,
            "hooray": 0,
            "confused": 0,
            "heart": 0,
            "rocket": 0,
            "eyes": 0
        },
        "performed_via_github_app": null
    }
]